src/pages/instances/actions/StopInstanceBtn.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { useQueryClient } from "@tanstack/react-query";
+import { stopInstance } from "api/instances";
+import { queryKeys } from "util/queryKeys";
+import ConfirmationButton from "components/ConfirmationButton";
+import { CheckboxInput } from "@canonical/react-components";
+import { useNotify } from "context/notify";
+import { useInstanceLoading } from "context/instanceLoading";
+import InstanceLink from "pages/instances/InstanceLink";
+
+interface Props {
+  instance: LxdInstance;
+}
+
+const StopInstanceBtn: FC<Props> = ({ instance }) => {
+  const instanceLoading = useInstanceLoading();
+  const notify = useNotify();
+  const [isForce, setForce] = useState(false);
+  const queryClient = useQueryClient();
+  const isLoading =
+    instanceLoading.getType(instance) === "Stopping" ||
+    instance.status === "Stopping";
+
+  const handleStop = () => {
+    instanceLoading.setLoading(instance, "Stopping");
+    stopInstance(instance, isForce)
+      .then(() => {
+        notify.success(
+          <>
+            Instance <InstanceLink instance={instance} /> stopped.
+          </>
+        );
+      })
+      .catch((e) => {
+        notify.failure(
+          <>
+            Error instance stop for <InstanceLink instance={instance} />.
+          </>,
+          e
+        );
+      })
+      .finally(() => {
+        instanceLoading.setFinish(instance);
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.instances],
+        });
+      });
+  };
+
+  return (
+    <ConfirmationButton
+      toggleAppearance="base"
+      isLoading={isLoading}
+      iconClass="p-icon--stop-instance"
+      title="Confirm stop"
+      confirmationMessage={`Are you sure you want to stop instance "${instance.name}"?`}
+      confirmationExtra={
+        <span className="u-float-left">
+          <CheckboxInput
+            inline
+            label="Force stop"
+            tabIndex={-1}
+            defaultChecked={isForce}
+            onClick={() => setForce((prev) => !prev)}
+          />
+        </span>
```

Comment:
This is used in 2 places, maybe it could be good to extract it to a dedicated component? So that we only need to pass 3 props: the `label`, the `defaultChecked` and the click handler - possibly 2 props only if we pass the whole `force` state...